mapbox/libandroid-telemetry/src/main/java/com/mapbox/services/android/telemetry/MapboxTelemetry.java

```diff
@@ -652,6 +682,18 @@ private void pushTurnstileEvent() {
     event.put(MapboxEvent.KEY_USER_ID, mapboxVendorId);
     event.put(MapboxEvent.KEY_ENABLED_TELEMETRY, isTelemetryEnabled());
 
+    if (sdkIdentifier == null) {
```

Comment:
What about simplifying this block of code (`if else`) extracting it into a different method and work to make the intention clearer?

Reply:
We should send `null` values when not calling the initialize method that uses `sdkIdentifier` and `sdkVersion` explicitly (when `sdkIdentifier` and `sdkVersion` are empty). Right now, this condition is always false because we initialize `sdkIdentifier` and `sdkVersion` to `""` and if any of them are `null` there is a previous check which will throw a `TelemetryException`.